Lesson_07/homework_07.py

```diff
@@ -0,0 +1,251 @@
+def average_value(numbers_list):
+    """Average value of numbers in list.
+
+    Args:
+        numbers_list (list): List of numbers
+
+    Returns:
+        float: List of numbers average value
+    """
+    return sum(numbers_list) / len(numbers_list)
+
+
+numbers = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+average_result = average_value(numbers)
+
+
+# task 4
+# Написати функцію, яка приймає рядок та повертає його у зворотному порядку.
+def reversed_string(test_string):
+    """Reverse somme string.
+
+    Args:
+        test_string (string): Some text
+
+    Returns:
+        string: Reversed string
+    """
+    return test_string[::-1]
+
+
+txt = 'Hello World!'
+reversed_txt = reversed_string(txt)
+
+
+# task 5
+# Написати функцію, яка приймає список слів
+# та повертає найдовше слово у списку.
+def max_word(word_list):
+    """Find word with maximum length from the list.
+
+    Args:
+        word_list (list): List of words
+
+    Returns:
+        string: Word with maximum length
+    """
+    return max(str(word_list), key=len)
+
+
+words = ['automobile', 'hi', 'pencil', 'Hello', 1]
+word_result = max_word(words)
+
+
+# task 6
+# Написати функцію, яка приймає два рядки та повертає індекс першого входження
+# другого рядка у перший рядок, якщо другий рядок є підрядком першого рядка,
+# та -1, якщо другий рядок не є підрядком першого рядка.
+def find_substring(str1, str2):
+    """First entry of the second line into the first line.
+
+    Args:
+        str1 (string): First string
+        str2 (string): Second string
+
+    Returns:
+        int: Index of the first entry
+    """
+    if str2 in str1:
+        return str1.find(str2[0])
+    return -1
+
+
+string1 = 'Hello, world!'
+string2 = 'world'
+print(find_substring(string1, string2))  # поверне 7
+
+string1 = 'The quick brown fox jumps over the lazy dog'
+string2 = 'cat'
+print(find_substring(string1, string2))  # поверне -1
+
+
+# Оберіть будь-які 4 таски з попередніх домашніх робіт та
+# перетворіть їх у 4 функції, що отримують значення та повертають результат.
+# Обоязково документуйте функції та дайте зрозумілі імена змінним.
+# task 7
+# Calculate unique symbols in the string.
+def unique_symbols_quantity(text):
+    """Calculate unique symbols in text.
+
+    Args:
+        text (string): Some text
+
+    Returns:
+        int: Quantity of unique symbols
+    """
+    return len(set(text.lower().replace(' ', '')))
+
+
+entered_text = 'The quick brown fox jumps over the lazy dog'
+unique_symbols = unique_symbols_quantity(entered_text)
+
+
+# task 8
+# Calculate sum of all Even number in list 'entered_numbers.
+def even_number(numbers_list):
+    """Calculate sum of all even numbers.
+
+    Args:
+        numbers_list (list): List of numbers.
+
+    Returns:
+        int: Sum of even numbers.
+    """
+    even_numbers_sum = 0
+    for number in numbers_list:
+        if number % 2 == 0:
+            even_numbers_sum += number
+    return even_numbers_sum
```

Comment:
```suggestion
    even_numbers_sum = 0
    for number in numbers_list:
        if number % 2 == 0:
            even_numbers_sum += number
    return even_numbers_sum
```
Please try to use list comprehension here + sum()

Reply:
Updated in commit: f61bf23d85eca83d002c0c3ff63679065222887d

`    return sum(number for number in numbers_list if number % 2 == 0)
`